File: c#/BigEgg.Algorithm/BigEgg.Algorithm/Collections/IBag.cs
using System.Collections.Generic;

namespace BigEgg.Algorithm.Collections
{
    public interface IBag<T> : IEnumerable<T>
    {
        /// <summary>
        /// Is this bag empty?
        /// </summary>
        /// <returns><c>True</c> if this bag is empty; <c>false</c> otherwise</returns>
        bool IsEmpty();

        /// <summary>
        /// Returns the number of items in the bag.
        /// </summary>
        /// <returns>The number of items in the bag.</returns>
        int Size();

        /// <summary>
        /// Adds the item to this bag.
        /// </summary>
        /// <param name="item">The item to add.</param>
        /// <exception cref="System.ArgumentNullException">Item cannot be null.</exception>
        void Add(T item);
    }
}

File: java/Algorithms/src/Collections/IBag.java
package Collections;

/**
 * Created by jianming.xiao on 10/11/14.
 */
public interface IBag<Item> extends Iterable<Item> {
    /**
     * Is this bag empty?
     *
     * @return true if this bag is empty; false otherwise
     */
    boolean isEmpty();

    /**
     * Returns the number of items in this bag.
     *
     * @return the number of items in this bag
     */
    int size();

    /**
     * Adds the item to this bag.
     *
     * @param item the item to add to this bag
     * @throws NullPointerException if the <tt>item</tt> is null;
     */
    void add(Item item);
}
